### DISCUSSION TITLE
Support for Merging Cells in Tables

### DISCUSSION THREAD

**Discussion by mantonie** [squidfunk/mkdocs-material#4845] [Q&A] [answered]
Any chance merging cells would be supported in MkDocs Material? 

I know Markdown does not support more complex usage of tables... is there any way to workaround it other than writing HTML code (or plans to provide this kind of support)?

Thanks, appreciate all the answers in advance!

**Comment by squidfunk** [comment, marked as answer]
Thanks for suggesting. As you noted, it's not supported by Python Markdown, so if you don't want to use HTML, another third-party Markdown extension would need to be created (or support added to Python Markdown). It's beyond the scope of this project, but it's definitely an excellent use case for a third-party extension! We could also advertise such an extension on our docs.

**Reply by mantonie** [reply]
Makes sense. Thanks for very quick answer!

**Comment by gledos** [comment]
The Python Markdown used by MkDocs has no way of doing this.

But there is a more convenient tool than directly editing HTML tables, which is to use [HTML Tables generator](https://www.tablesgenerator.com/html_tables).

![圖片](https://user-images.githubusercontent.com/23720620/213176815-96751dfc-0a02-49c3-ac81-bb25edbbacf3.png)

**Comment by gchongo** [comment]
Using this tool [HTML Tables generator](https://www.tablesgenerator.com/html_tables) indeed makes it convenient to create complex tables, merge cells using HTML, etc. 
However, there's an issue. I tried using CSS to define the inner and outer borders of the table and to center the text in the cells, but it didn't work. Do you have any suggestions? Here is the relevant code.
This table is the effect I want to achieve, with inner and outer borders and centered text.
![image](https://github.com/user-attachments/assets/13bfb728-be86-4569-9de5-b48d4b0563c5)
However, the code is as follows, writing the style in the table is too complicated.
`<div class="tg-wrap"><table>
  <thead>
    <tr>
      <th colspan="6" style="border: 1px solid #000; padding: 8px; text-align: center;">Results</th>
      <th style="border: 1px solid #000; padding: 8px; text-align: center;">s</th>
      <th colspan="7" style="border: 1px solid #000; padding: 8px; text-align: center;">ssas</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">No</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Competition</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">John</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Adam</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Robert</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Paul</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">No</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Competition</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">No</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Competition</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">John</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Adam</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Robert</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Paul</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Swimming</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1:30</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">2:05</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1:15</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1:41</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Swimming</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Swimming</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1:30</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">2:05</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1:15</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">1:41</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">2</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Running</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">15:30</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">14:10</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">15:45</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">16:00</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">2</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Running</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">2</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Running</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">15:30</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">14:10</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">15:45</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">16:00</td>
    </tr>
    <tr>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">3</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Shooting</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">70%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">55%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">90%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">88%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">3</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Shooting</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">3</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">Shooting</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">70%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">55%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">90%</td>
      <td style="border: 1px solid #000; padding: 8px; text-align: center;">88%</td>
    </tr>
  </tbody>
</table></div>`

Below is the method using custom CSS. This is the code in YAML.
`table {
  border-collapse: collapse;
  width: 100%;
}

th, td {
  border: 1px solid #000; /* black */
  padding: 8px;
  text-align: center;
}`

and
`extra_css:
  - css/custom.css`

However, the final effect is as follows, and it seems the CSS did not take effect.
![image](https://github.com/user-attachments/assets/92c8f3be-9fb4-4445-b854-29651277b502)

**Reply by gchongo** [reply]
![image](https://github.com/user-attachments/assets/ff950b23-41c9-4ac5-bc1d-9eb4dd8ff342)
